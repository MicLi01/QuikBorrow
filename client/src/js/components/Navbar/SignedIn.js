import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { logoutUser } from '../../actions';
import { Toolbar, Avatar } from '@material-ui/core';
import styled from 'styled-components'
import AddItemModal from '../AddItemModal/AddItemModal';

const UnstyledLink = styled(Link)`
    color: inherit;
    text-decoration: inherit;
    margin-left: 20px;
`

const SignedIn = (props) => {
    const onClick = e => {
        e.preventDefault();
        props.logoutUser();
    };
    const userProfileId = props.id 

    const [ openModal, setOpenModal ] = useState(false)

    return (
        <Toolbar>
            <AddItemModal open={openModal} onClose={e => setOpenModal(false)} />
            
            <UnstyledLink onClick={e => setOpenModal(true)}>Add Item</UnstyledLink>
            <UnstyledLink to='/products'>Products</UnstyledLink>
            <UnstyledLink to={ '/profile/'+userProfileId}><Avatar src='https://avatars1.githubusercontent.com/u/619960?s=460&v=4'/> </UnstyledLink>
            <UnstyledLink to='#' onClick={onClick}>Sign Out</UnstyledLink>
        </Toolbar>
    )
}

export default connect(null, { logoutUser })(SignedIn);
