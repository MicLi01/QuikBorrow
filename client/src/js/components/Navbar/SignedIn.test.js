import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';
import { SignedIn } from './SignedIn';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'


Enzyme.configure({ adapter: new Adapter() });
const createStore = configureStore()

describe('SignedIn', () => {

  	// it('Test click event', () => {
	// 	const mockCallBack = jest.fn();
	// 	const store = createStore({ auth: { user: {} } })

    // 	const tree = mount(
	// 		<Provider store={store}>
	// 		<SignedIn onClick={mockCallBack}/>
	// 		</Provider>
	// 		)
	// 	tree.simulate('click');
    // 	expect(mockCallBack.mock.calls.length).toEqual(1);
	//   });
    it('render SignedIn', () => {
        const store = createStore({ auth: { user: {} } })

        const component = renderer.create(
                <Provider store={store}>
                    <MemoryRouter>
                        <SignedIn />
                    </MemoryRouter>
                </Provider>
			);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
        });

});
