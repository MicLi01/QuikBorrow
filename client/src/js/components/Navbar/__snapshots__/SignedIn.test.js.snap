// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignedIn render SignedIn 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
>
  <a
    className="sc-ifAKCX joNUkC"
    href="/"
    onClick={[Function]}
  >
    Add Item
  </a>
  <a
    className="sc-ifAKCX joNUkC"
    href="/products"
    onClick={[Function]}
  >
    Products
  </a>
  <a
    className="sc-ifAKCX joNUkC"
    href="/profile/undefined"
    onClick={[Function]}
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle"
    >
      <img
        className="MuiAvatar-img"
        src="https://avatars1.githubusercontent.com/u/619960?s=460&v=4"
      />
    </div>
     
  </a>
  <a
    className="sc-ifAKCX joNUkC"
    href="/"
    onClick={[Function]}
  >
    Sign Out
  </a>
</div>
`;
